lib/ansible/modules/cloud/vmware/vmware_guest_serial_port.py

```diff
@@ -0,0 +1,458 @@
+                                          " product versioning restrictions: %s" % to_native(e.msg))
+        if task.info.state == 'error':
+            results = {'changed': self.change_applied, 'failed': True, 'msg': task.info.error.msg}
+        else:
+            serial_port_info = get_serial_port_info(vm_obj)
+            results = {'changed': self.change_applied, 'failed': False, 'serial_port_info': serial_port_info}
+
+        return results
+
+
+def get_serial_port(vm_obj, backing):
+    """
+    Return the serial port of specified backing type
+    """
+    serial_port = None
+    backing_type_mapping = {
+        'network': vim.vm.device.VirtualSerialPort.URIBackingInfo,
+        'pipe': vim.vm.device.VirtualSerialPort.PipeBackingInfo,
+        'device': vim.vm.device.VirtualSerialPort.DeviceBackingInfo,
+        'file': vim.vm.device.VirtualSerialPort.FileBackingInfo
+    }
+    for device in vm_obj.config.hardware.device:
+        if isinstance(device, vim.vm.device.VirtualSerialPort):
+            if isinstance(device.backing, backing_type_mapping[backing['type']]):
+                serial_port = device
+                break
+
+    return serial_port
+
+
+def get_serial_port_info(vm_obj):
+    """
+    Get the serial port info
+    """
+    serial_port_info = dict()
+    if vm_obj is None:
+        return serial_port_info
+    port_index = 0
+    for port in vm_obj.config.hardware.device:
+        backing = dict()
+        if isinstance(port, vim.vm.device.VirtualSerialPort):
+            if isinstance(port.backing, vim.vm.device.VirtualSerialPort.URIBackingInfo):
+                backing['backing_type'] = 'network'
+                backing['direction'] = port.backing.direction
+                backing['service_uri'] = port.backing.serviceURI
+            elif isinstance(port.backing, vim.vm.device.VirtualSerialPort.PipeBackingInfo):
+                backing['backing_type'] = 'pipe'
+                backing['pipe_name'] = port.backing.pipeName
+                backing['endpoint'] = port.backing.endpoint
+                backing['no_rx_loss'] = port.backing.noRxLoss
+            elif isinstance(port.backing, vim.vm.device.VirtualSerialPort.DeviceBackingInfo):
+                backing['backing_type'] = 'device'
+                backing['device_name'] = port.backing.deviceName
+            elif isinstance(port.backing, vim.vm.device.VirtualSerialPort.FileBackingInfo):
+                backing['backing_type'] = 'file'
+                backing['file_path'] = port.backing.fileName
+            else:
+                continue
+            serial_port_info[port_index] = backing
+            port_index += 1
+    return serial_port_info
+
+
+def create_serial_port(backing):
+    """
+    Create a new serial port
+    """
+    serial_spec = vim.vm.device.VirtualDeviceSpec()
+    serial_port = vim.vm.device.VirtualSerialPort()
+    serial_port.yieldOnPoll = True
```

Comment:
I maybe wrong but we are getting user input for `yield_on_poll` but you are not using here.